contribs/src/main/java/com/netflix/conductor/contribs/publisher/RestClientManager.java

```diff
@@ -0,0 +1,151 @@
+package com.netflix.conductor.contribs.publisher;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.ClientProtocolException;
+import org.apache.http.client.HttpRequestRetryHandler;
+import org.apache.http.client.config.RequestConfig;
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpPost;
+import org.apache.http.entity.StringEntity;
+import org.apache.http.impl.client.CloseableHttpClient;
+import org.apache.http.impl.client.HttpClients;
+import org.apache.http.impl.conn.PoolingHttpClientConnectionManager;
+import org.apache.http.protocol.HttpContext;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import javax.inject.Singleton;
+import java.io.IOException;
+import java.util.HashMap;
+import java.util.Map;
+
+/*
+
+https://www.baeldung.com/httpclient-timeout
+the Connection Timeout (http.connection.timeout) – the time to establish the connection with the remote host
+the Socket Timeout (http.socket.timeout) – the time waiting for data – after establishing the connection; maximum time of inactivity between two data packets
+the Connection Manager Timeout (http.connection-manager.timeout) – the time to wait for a connection from the connection manager/pool
+
+ https://learnbyinsight.com/2020/06/29/httpclient-single-instance-or-multiple/
+
+ https://www.programcreek.com/java-api-examples/?api=org.apache.http.impl.conn.PoolingHttpClientConnectionManager
+ */
+@Singleton
+public class RestClientManager {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RestClientManager.class);
+
+    enum NotificationType {
+        TASK,
+        WORKFLOW
+    };
+    private static final String TASK_NOTIFICATION_ENDPOINT = "workflow/TaskNotifications";
+    private static final String WORKFLOW_NOTIFICATION_ENDPOINT = "workflow/WorkflowNotifications";
+    private static final String URL = "http://bullwinkle.default.svc.cluster.local:7979/v1";
+
+    private static final int DEFAULT_CONNECT_TIMEOUT = 10;
+    private static final int DEFAULT_REQUEST_TIMEOUT = 10;
+    private static final int DEFAULT_SOCKET_TIMEOUT = 10;
```

Comment:
To be clear, these names can be changed to reflect the time unit.  for example, DEFAULT_CONNECT_TIMEOUT _IN_SECONDS.  Since the API takes millisecs as argument, explicit naming will help with readability.

Reply:
Done